packages/kbn-test/src/es/test_es_cluster.ts

```diff
@@ -139,3 +148,28 @@ export function createTestEsCluster(options: TestClusterFactoryOptions) {
     }
   })();
 }
+
+/**
+ * Takes in two arrays of 'key=val' strings, allowing values from the second array
+ * to override values in the first.
+ *
+ * @example
+ *
+ * mergeDefaultArgs(['foo=a', 'bar=b'], ['foo=c', 'baz=d']); // returns ['foo=c', 'bar=b', 'baz=d']
+ */
+function mergeDefaultArgs(defaultArgs: string[], customArgs: string[]): string[] {
+  const toArgsObject = (args: string[]) => {
+    const obj: Record<string, string> = {};
+
+    args.forEach((arg) => {
+      const [key, val] = arg.split('=');
+      obj[key] = val;
+    });
+
+    return obj;
+  };
+
+  return Object.entries({ ...toArgsObject(defaultArgs), ...toArgsObject(customArgs) }).map(
+    ([key, val]) => `${key}=${val}`
+  );
+}
```

Comment:
In addition to the changes here, we'll need to update `es_test_config` to support multiple nodes. (Currently `getUrl` returns one URL, which can be read from the `TEST_ES_URL` environment variable).

Wasn't sure what the preferred way to do this was (more environment variables? something else?). Will need a bit of thought.